src/analysis/calculus/series.lean

```diff
@@ -0,0 +1,311 @@
+  rw [dist_eq_norm, ← sum_add_tsum_subtype_compl (summable_of_summable_norm A) t, add_sub_cancel'],
+  apply lt_of_le_of_lt _ ht,
+  apply (norm_tsum_le_tsum_norm (A.subtype _)).trans,
+  exact tsum_le_tsum (λ n, hfu _ _ hx) (A.subtype _) (hu.subtype _)
+end
+
+/-- An infinite sum of functions with summable sup norm is the uniform limit of its partial sums.
+Version relative to a set, with index set `ℕ`. -/
+lemma tendsto_uniformly_on_tsum_nat {f : ℕ → β → F} {u : ℕ → ℝ} (hu : summable u) {s : set β}
+  (hfu : ∀ n x, x ∈ s → ∥f n x∥ ≤ u n) :
+  tendsto_uniformly_on (λ N, (λ x, ∑ n in finset.range N, f n x)) (λ x, ∑' n, f n x) at_top s :=
+λ v hv, tendsto_finset_range.eventually (tendsto_uniformly_on_tsum hu hfu v hv)
+
+/-- An infinite sum of functions with summable sup norm is the uniform limit of its partial sums.
+Version with general index set. -/
+lemma tendsto_uniformly_tsum {f : α → β → F} {u : α → ℝ} (hu : summable u)
+  (hfu : ∀ n x, ∥f n x∥ ≤ u n) :
+  tendsto_uniformly (λ (t : finset α), (λ x, ∑ n in t, f n x)) (λ x, ∑' n, f n x) at_top :=
+by { rw ← tendsto_uniformly_on_univ, exact tendsto_uniformly_on_tsum hu (λ n x hx, hfu n x) }
+
+/-- An infinite sum of functions with summable sup norm is the uniform limit of its partial sums.
+Version with index set `ℕ`. -/
+lemma tendsto_uniformly_tsum_nat {f : ℕ → β → F} {u : ℕ → ℝ} (hu : summable u)
+  (hfu : ∀ n x, ∥f n x∥ ≤ u n) :
+  tendsto_uniformly (λ N, (λ x, ∑ n in finset.range N, f n x)) (λ x, ∑' n, f n x) at_top :=
+λ v hv, tendsto_finset_range.eventually (tendsto_uniformly_tsum hu hfu v hv)
+
+/-- An infinite sum of functions with summable sup norm is continuous on a set if each individual
+function is. -/
+lemma continuous_on_tsum [topological_space β]
+  {f : α → β → F} {s : set β} (hf : ∀ i, continuous_on (f i) s) {u : α → ℝ} (hu : summable u)
+  (hfu : ∀ n x, x ∈ s → ∥f n x∥ ≤ u n) :
+  continuous_on (λ x, ∑' n, f n x) s :=
+begin
+  classical,
+  refine (tendsto_uniformly_on_tsum hu hfu).continuous_on (eventually_of_forall _),
+  assume t,
+  exact continuous_on_finset_sum _ (λ i hi, hf i),
+end
+
+/-- An infinite sum of functions with summable sup norm is continuous if each individual
+function is. -/
+lemma continuous_tsum [topological_space β]
+  {f : α → β → F} (hf : ∀ i, continuous (f i)) {u : α → ℝ} (hu : summable u)
+  (hfu : ∀ n x, ∥f n x∥ ≤ u n) :
+  continuous (λ x, ∑' n, f n x) :=
+begin
+  simp_rw [continuous_iff_continuous_on_univ] at hf ⊢,
+  exact continuous_on_tsum hf hu (λ n x hx, hfu n x),
+end
+
+
+/-! ### Differentiability -/
+
+variables [normed_space 𝕜 F]
+
+/-- Consider a series of functions `∑' n, f n x` on a preconnected open set. If the series converges
+at a point, and all functions in the series are differentiable with a summable bound on the
+derivatives, then the series converges everywhere on the set. -/
+lemma summable_of_summable_has_fderiv_at_of_is_preconnected
+  {f : α → E → F} {f' : α → E → (E →L[𝕜] F)} {u : α → ℝ} (hu : summable u)
```

Comment:
At least the first three variables and `{N : ℕ∞}` should be move outside of the lemmas, since they are in all following lemmas.